Xamarin.PropertyEditing/ViewModels/PropertyGroupViewModel.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Xamarin.PropertyEditing.ViewModels
+{
+	internal class PropertyGroupViewModel
+		: EditorViewModel, IFilterable
+	{
+		public PropertyGroupViewModel (string category, IEnumerable<PropertyViewModel> properties, IEnumerable<IObjectEditor> objEditors)
+			: base (objEditors)
+		{
+			if (category == null)
+				throw new ArgumentNullException (nameof(category));
+			if (properties == null)
+				throw new ArgumentNullException (nameof (properties));
+
+			Category = category;
+
+			this.properties = properties.ToArray ();
+			foreach (var vm in this.properties) {
+				if (vm.IsAvailable)
+					this.propertiesView.Add (vm);
+
+				vm.PropertyChanged += OnChildPropertyChanged;
+			}
+		}
+
+		public IReadOnlyList<PropertyViewModel> Properties => this.propertiesView;
+
+		public override string Name
+		{
+			get;
+		}
+
+		public override string Category
+		{
+			get;
+		}
+
+		public bool HasChildElements => (this.propertiesView.Count > 0);
+
+		public string FilterText
+		{
+			get { return this.filterText; }
+			set
+			{
+				if (this.filterText == value)
+					return;
+
+				string oldFilter = this.filterText;
+				this.filterText = value;
+				Filter (oldFilter);
+				OnPropertyChanged();
+			}
+		}
+
+		private readonly PropertyViewModel[] properties;
+		private readonly ObservableCollectionEx<PropertyViewModel> propertiesView = new ObservableCollectionEx<PropertyViewModel> ();
+		private string filterText;
+
+		private void Filter (string oldFilter)
+		{
+			if (FilterText != null && (String.IsNullOrWhiteSpace (oldFilter) || FilterText.StartsWith (oldFilter, StringComparison.OrdinalIgnoreCase))) {
+				var current = new List<PropertyViewModel> (this.propertiesView);
+				for (int i = 0; i < current.Count; i++) {
+					var vm = current[i];
+					if (!MatchesFilter (vm))
+						this.propertiesView.Remove (vm);
+				}
+			} else {
+				this.propertiesView.Reset (this.properties.Where (MatchesFilter).OrderBy (p => p.Property.Name));
+			}
+
+			OnPropertyChanged (nameof(HasChildElements));
```

Comment:
Won't this fire even if you go from a filter that yields zero children to another that also yields none, and vice-versa?

Reply:
Technically, but it was the less error prone route with minimal impact. I'll see about changing it.